### ISSUE TITLE
Firelands items are no longer being tracked properly after the difficulties were merged

### ISSUE THREAD

**Issue by SacredDuckwhale** [SacredDuckwhale/Rarity#162]
* Firelands tracking is most probably broken since the difficulty was changed
* Potentially also affects Black Temple? (Needs testing)
* Any other raids that were done dirty like that?

**Comment by SacredDuckwhale** [comment]
I can confirm that the difficulty setting changed and the Firelands raid is now dynamically adjusted. Testing yields for HEROIC:

* Shannox didn't work
* Ragnaros didn't work, even after removing the old instanceDifficulties settings (does this break anything?)

And for NORMAL:

* TBD

**Comment by SacredDuckwhale** [comment]
It's probably easiest to just use the existing encounter detection, or looting the NPCs directly (should work for all bosses except Ragnaros).

**Comment by SacredDuckwhale** [comment]
Did another run where literally no statistics worked, at all (new toon that had never done FL). Heroic difficulty.

Seems like Blizzard killed the statistics again, but since they SOMETIMES work I can't just add an alternative encounter-based detection method or there will be duplicate attempts.

Therefore, the workaround I added (in 14f017d4a135a370e7d673082f781d42dd287097) is to switch all bosses that drop pets to a NPC-based method, which should always work and also has the benefit of avoiding incorrect counts taken from kill statistics (pets were added retroactively).

Ragnaros and Alysrazor still use the statistics-based detection to hopefully update to the correct values when/if Blizzard fixes it, and because using different methods would cause issues with lost progress (statistics are usually more reliable than data cached by the addon, if they work...) for people who reset their addons etc.

**Comment by brett1337** [comment]
Has blizzard never fixed this? So it is impossible to track Flametalon of Alysrazor attempts?

**Comment by Duckwhale** [comment]
> Has blizzard never fixed this? So it is impossible to track Flametalon of Alysrazor attempts?

I haven't done any testing since my last post. If statistics aren't updating in the achievements UI, then no, they clearly haven't.

It's not impossible to track attempts. But it's (probably) very difficult to track them without kill statistics in a way that doesn't 

* break backwards compatibility (i.e., keeps and uses the data for previous attempts)
* cause issues if switched back to kill statistics once (if...) Blizzard fixes the problem
* involve sweeping changes to the codebase that are likely to cause additional problems
* increase the complexity of the code (and therefore mental burden on the few people working with it)

In short: Sounds like a headache to me. I'd rather that Blizzard fix their game, they have plenty of resources while I do not.

---

As a workaround, I can imagine adding the two relevant mounts as a custom item might work (using NPC IDs, not statistics).

**Comment by brett1337** [comment]
That makes a lot of sense to me, I'll do that and submit a bug report to blizzard for whatever that's worth!
